### PR TITLE
Cassidy Avery -- working w/o stretch

### PR DESCRIPTION
**Pull request by tacolim**

### CODE REVIEW

File: Answers.md [hunk 1]
```diff
@@ -1 +1,16 @@
 <!-- Answers to the Short Answer Essay Questions go here -->
+## Describe Middleware, Sessions (as we know them in express), bcrypt and JWT.
+Middleware
```
Comment (ryanhca) [inline Answers.md]: wat?

File: Answers.md [hunk 2]
```diff
@@ -1 +1,16 @@
...
+Sessions
+bcrypt
+JWT
+
+## What does bcrypt do in order to prevent attacks?
+bcrypt makes it very difficult for someone to use brute force to figure out passwords
```
Comment (ryanhca) [inline Answers.md]: Nice! this is a textbook answer.

File: api/utils/middlewares.js [hunk 3]
```diff
@@ -5,6 +5,15 @@ const User = require('../models/userModels');
 const { mysecret } = require('../../config');
 const SaltRounds = 11;
 
+const sendError = (err, res) => {
+  res.status(422);
+  if (err && err.message) {
+    res.json({ message: err.message, stack: err.stack });
```
Comment (ryanhca) [inline api/utils/middlewares.js]: Nice stack tracing!

File: api/utils/middlewares.js [hunk 4]
```diff
@@ -27,6 +36,19 @@ const encryptUserPW = (req, res, next) => {
   // Once the password is encrypted using bcrypt, you'll need to save the user the DB.
   // Once the user is set, take the savedUser and set the returned document from Mongo on req.user
   // call next to head back into the route handler for encryptUserPW
+  if (!password) {
+    sendError('Password Required', res);
+    return;
+  }
+  bcrypt
+    .hash(password, SaltRounds)
+    .then((pw) => {
```
Comment (ryanhca) [inline api/utils/middlewares.js]: Don't need those parens around a single parameter but /shrug you're good to leave them :)

File: Answers.md (hunk 1 above)
Comment (tacolim) [reply Answers.md]: whoops! I started answers.md, then got sidetracked and saved and forgot to come back haha
I'll finish these up this weekend.

**Pull request closed without merging**